src/gui/inputdevices/TouchDrawingInputHandler.cpp

```diff
@@ -72,8 +73,11 @@ auto TouchDrawingInputHandler::handleImpl(InputEvent const& event) -> bool {
 
 auto TouchDrawingInputHandler::changeTool(InputEvent const& event) -> bool {
     Settings* settings = this->inputContext->getSettings();
-    ButtonConfig* cfgTouch = settings->getTouchButtonConfig();
+    ButtonConfig* cfgTouch = settings->getButtonConfig(Buttons::BUTTON_TOUCH);
     ToolHandler* toolHandler = this->inputContext->getToolHandler();
+    bool toolChanged = false;
+    bool pressed = false;
+    bool configChanged = true;
```

Comment:
It is bad style to force callers to reserve space for data they do not require. Add a minimized signature for `applyButton` instead.